packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1292,6 +1303,40 @@ export class TransactionController extends BaseController<
 
     return Common.custom(customChainParams);
   }
+
+  private generateDappSuggestedGasFee(
+    transaction: Transaction,
+    origin?: string,
+  ) {
+    let dappSuggestedGasFees: DappSuggestedGasFees | undefined;
+    // If we are dealing with a transaction suggested by a dapp and not
+    // an internally created metamask transaction, we need to keep record of
+    // the originally submitted gasParams.
+    if (typeof origin === 'string' && origin !== ORIGIN_METAMASK) {
```

Comment:
I appreciate you've copied this exactly from the extension, but is it worth simplifying the code whenever possible to make any future maintenance easier.

Could we add some early returns and remove the `typeof` checks here to end up with something like:

```
  if (!origin || origin == ORIGIN_METAMASK) {
    return undefined;
  }

  const { gasPrice, maxFeePerGas, maxPriorityFeePerGas, gas } = transaction;

  if (!gasPrice && !maxFeePerGas && !maxPriorityFeePerGas && !gas) {
    return undefined;
  }

  const dappSuggestedGasFees = {};

  if (gasPrice) {
    dappSuggestedGasFees.gasPrice = gasPrice;
  } else if (maxFeePerGas || maxPriorityFeePerGas) {
    dappSuggestedGasFees.maxFeePerGas = maxFeePerGas;
    dappSuggestedGasFees.maxPriorityFeePerGas = maxPriorityFeePerGas;
  }

  if (gas) {
    dappSuggestedGasFees.gas = gas;
  }

  return dappSuggestedGasFees;
```

Reply:
Thanks for the suggestion, done in https://github.com/MetaMask/core/pull/1617/commits/fc0409f783f018d390edf376a587f2006b62335b